hyppo/ksample/energy.py

```diff
@@ -106,16 +107,18 @@ def statistic(self, x, y):
         stat : float
             The computed Energy statistic.
         """
-        distx = x
-        disty = y
+        dataMatrix = np.concatenate([x, y])
+        labelMatrix = np.concatenate([np.zeros(len(x)), np.ones(len(y))])
         n = x.shape[0]
         m = y.shape[0]
-
-        distx, disty = compute_dist(x, y, metric=self.compute_distance, **self.kwargs)
+        
+        dcov = dcor.distance_covariance_sqr(dataMatrix, labelMatrix)
 
         # exact equivalence transformation Dcorr and Energy
         stat = (
-            _dcov(distx, disty, self.bias) * (2 * (n**2) * (m**2)) / ((n + m) ** 4)
```

Comment:
This line is really the only one that needs to be changed based on the https://link.springer.com/article/10.1007/s10182-020-00378-1 and the Nonpar Manova via independence testing paper